### PR TITLE
Consolidate scripts and python package

### LINKED ISSUE #169: Machine readable model property overview
Would be nice to have the table provided in README.md in some machine readable format like `.csv` or `.tsv`.

### LINKED ISSUE #232: Extend overview table: priors
It would be nice to see in the overview table which problems use `objectivePrior*`. Either yes/no or which types.

### PR DESCRIPTION
**Pull request by dweindl**
* some functionality from `scripts/` would be useful inside the library
* some functionality from the library was replicated in `scripts/`

-> consolidate (see also #208), turn separate scripts into entrypoints (currently prefixed with `bmp-` from Benchmark-Models-Petab)

* Reduce number of package configuration files
* replace isort/flake/... by ruff
* Problem feature table accessible through the library (closes #169)
* `--help` for the overview table entrypoint
* Extend overview table: add objective priors (Closes #232

### CODE REVIEW

File: src/python/pyproject.toml [hunk 1]
```diff
@@ -2,13 +2,44 @@
...
+
+[project]
+name = "benchmark_models_petab"
+description = "A collection of models with experimental data in the PEtab format"
+dynamic = ["version"]
+readme = { file = "README.md", content-type = "text/markdown" }
+requires-python = ">=3.10"
+license = { file = "LICENSE" }
+authors = [
+    { name = "The Benchmark-Models-PEtab maintainers", email = "yannik.schaelte@gmail.com" }
```
Comment (dilpath) [inline src/python/pyproject.toml:18]: You're welcome to add yourself here, as you like.

**Review by dilpath — APPROVED** [review]
Thanks for the many improvements! I checked the `bmp-*` commands locally :+1:

Could add `from .overview import get_overview_df` to `__init__`, and for the other commands too, but fine as-is for me.

File: src/python/pyproject.toml (hunk 1 above)
Comment (dweindl) [reply src/python/pyproject.toml]: Updated maintainers/authors.

**Comment by dweindl** [comment]
> Could add `from .overview import get_overview_df` to `__init__`, and for the other commands too, but fine as-is for me. At the moment there is no documentation, but if it's in `__init__` then it's a bit more findable.

Did it for `get_overview_df`. The rest can wait until after further cleanup and better documentation.

**Pull request merged**